### ISSUE TITLE
Custom Content Security Policies not being built correctly

### ISSUE THREAD

**Issue by spSlaine** [GaProgMan/OwaspHeaders.Core#51]
When using a custom configuration to determine the Content Security Policy e.g.

``` csharp

public void Configure(IApplicationBuilder app, IHostingEnvironment env)
{
    ...
    app.UseSecureHeadersMiddleware(CustomConfiguration());
    ...
}

private static SecureHeadersMiddlewareConfiguration CustomConfiguration()
{

    var safeScripts = new List<ContentSecurityPolicyElement>
    {
        ContentSecurityPolicyHelpers.CreateSelfDirective(),
        new ContentSecurityPolicyElement()
        {
            CommandType = CspCommandType.Uri, 
            DirectiveOrUri = "https://safeaddress.com/"
         }
    };

            var safeStyles = new List<ContentSecurityPolicyElement>
            {
                ContentSecurityPolicyHelpers.CreateSelfDirective(),
                new ContentSecurityPolicyElement()
                {
                    CommandType = CspCommandType.Uri,
                    DirectiveOrUri = "https://safeaddress.com/"
                },
                new ContentSecurityPolicyElement()
                {
                    CommandType = CspCommandType.Uri, 
                    DirectiveOrUri = "https://fonts.googleapis.com/"
                }
            };


            var safeFonts = new List<ContentSecurityPolicyElement>
            {
                ContentSecurityPolicyHelpers.CreateSelfDirective(),
                new ContentSecurityPolicyElement()
                {
                    CommandType = CspCommandType.Uri,
                    DirectiveOrUri = "https://fonts.googleapis.com/"
                }
            };

            return SecureHeadersMiddlewareBuilder
                .CreateBuilder()
                .UseHsts()
                .UseXFrameOptions()
                .UseXSSProtection()
                .UseContentTypeOptions()
                .UseContentSecurityPolicy(null, true, true, null, null)
                .SetCspUris(safeScripts, CspUriType.Script)
                .SetCspUris(safeStyles, CspUriType.Style)
                .SetCspUris(safeFonts, CspUriType.Font)
                .UsePermittedCrossDomainPolicies()
                .UseReferrerPolicy()
                .Build();
        }
```
The Policy Elements that are marked as ```CspCommandType.Uri``` are being quoted as well being written as unquoted in the HEADER

e.g.

```
content-security-policy: script-src 'self' 'https://safeaddress.com/' https://safeaddress.com/;style-src 'self' 'https://safeaddress.com/' 'https://fonts.googleapis.com/' https://safeaddress.com/ https://fonts.googleapis.com/;font-src 'self' 'https://fonts.googleapis.com/' https://fonts.googleapis.com/;block-all-mixed-content; upgrade-insecure-requests;
```
I may be wrong but I understand the header should be returned with only the directives being quoted.

```
content-security-policy: script-src 'self' https://safeaddress.com/;style-src 'self' https://safeaddress.com/ https://fonts.googleapis.com/;font-src 'self'  https://fonts.googleapis.com/;block-all-mixed-content; upgrade-insecure-requests;
```

**Comment by GaProgMan** [comment]
Indeed you are correct about this. Only the directives (i.e. 'self') should be quoted. Strange that the rendered CSP still works in Chrome based browsers, though.

Then again, Chrome happily fixes many broken things automatically.

**Comment by GaProgMan** [comment]
Fixed in PR #53
